src/Quarrel/App.xaml.cs

```diff
@@ -42,15 +49,64 @@ namespace Quarrel
     /// </summary>
     sealed partial class App : Application
     {
+        public static IServiceProvider ServiceProvider { get; }
+        public static IConfiguration Configuration { get; }
+
+        public static ViewModelLocator ViewModelLocator => ServiceProvider.GetService<ViewModelLocator>();
+
+        static App()
+        {
+            var services = new ServiceCollection();
+
+            services.AddSingleton<ViewModelLocator, ViewModelLocator>();
+            services.AddLogging();
+
+            services.AddTransient<IConfiguration>(sp =>
+            {
+                var input = new Dictionary<string, string>
+                {
+                    {"Logging:LogLevel:Default", "Trace"},
+                    {"Logging:LogLevel:System", "Information"},
+                    {"Logging:LogLevel:Microsoft", "Information"},
+                };
+
+                IConfigurationBuilder configurationBuilder = new ConfigurationBuilder();
+                //configurationBuilder.AddJsonFile("appsettings.json");
+                configurationBuilder.AddInMemoryCollection(input);
+                return configurationBuilder.Build();
+            });
+
+            ServiceProvider = services.BuildServiceProvider();
+
+            Configuration = ServiceProvider.GetService<IConfiguration>();
+
+            var folder = ApplicationData.Current.LocalFolder;
+            string fullPath = $"{folder.Path}\\Logs\\App.log";
+
+            ServiceProvider.GetService<ILoggerFactory>().AddFile(fullPath, LogLevel.Debug);
+            ServiceProvider.GetService<ILoggerFactory>().AddDebug((s, l) => true);
+
```

Comment:
Why is the static constructor needed?

Reply:
By convention, Microsoft's IOC is initialized at the static context before the App is created.